lettermgmt/models/res_letter.py

```diff
@@ -1,161 +1,144 @@
-                                   ('cancel', 'Cancelled')],
-                                  'State', readonly=True,
-                                  track_visibility='onchange'),
-        'parent_id': fields.many2one('res.letter', 'Parent'),
-        'child_line': fields.one2many(
-            'res.letter', 'parent_id', 'Letter Lines'),
-        'channel_id': fields.many2one(
-            'letter.channel', 'Sent / Receive Source'),
-        'orig_ref': fields.char(
-            'Original Reference', help="Reference Number at Origin."),
-        'expeditor_ref': fields.char(
-            'Expeditor Reference', help="Reference Number used by Expeditor."),
-        'track_ref': fields.char(
-            'Tracking Reference', help="Reference Number used for Tracking."),
-        'weight': fields.float('Weight (in KG)'),
-        'size': fields.char('Size'),
-        'reassignment_ids': fields.one2many(
-            'letter.reassignment', 'letter_id', string='Reassignment lines',
-            help='Reassignment users and comments'),
-        'extern_partner_ids': fields.many2many(
-            'res.partner', string='Recipients'),
-    }
+    def default_sender(self):
+        move_type = self.env.context.get('move', False)
+        if move_type == 'out':
+            return self.env.user.company_id.partner_id
 
-    _defaults = {
-        'number': _get_number,
-        'snd_rec_date': fields.datetime.now,
-        'move': lambda self, cr, uid, context: context.get('move', 'in'),
-        'state': 'draft',
-    }
+    name = fields.Text('Subject', help="Subject of letter.")
+    folder_id = fields.Many2one(
+        'letter.folder', string='Folder',
+        help='Folder which contains letter.')
+    number = fields.Char(
+        'Number', help="Auto Generated Number of letter.",
+        default="/", required=True)
+    move = fields.Selection(
+        [('in', 'IN'), ('out', 'OUT')], 'Move', readonly=True,
+        help="Incoming or Outgoing Letter.",
+        default=lambda self: self.env.context.get('move', 'in'))
+    type_id = fields.Many2one(
+        'letter.type', string='Type',
+        help="Type of Letter, Depending upon size.")
+    category_ids = fields.Many2many(
+        'letter.category', string='Tags', help="Classification of Document.")
+    date = fields.Datetime('Letter Date', help='The letter\'s date')
+    snd_date = fields.Datetime(
+        'Sent / Received Date', defalut=fields.datetime.now,
+        help='Created Date of Letter Logging.')
+    rcv_date = fields.Datetime(
+        'Sent / Received Date', defalut=fields.datetime.now,
+        help='Created Date of Letter Logging.')
+    recipient_partner_id = fields.Many2one(
+        'res.partner', string='Recipient', track_visibility='onchange',
+        default=default_recipient)
+    sender_partner_id = fields.Many2one(
+        'res.partner', string='Sender', track_visibility='onchange',
+        default=default_sender)
+    note = fields.Text('Note')
```

Comment:
There's no need for this field string.
Although we could add some help to this field.

In #87 I had added this:

```py
    note = fields.Text(
        string='Delivery Notes',
        help='Indications for the delivery officer.')
```